### ISSUE TITLE
Cleanup removes available movies/series from DB

### ISSUE THREAD

**Issue by xadox-1st** [xbmc/xbmc#25724]
### Describe the bug
A cleanup of the library also removes movies and series from sources that are available.
If a scan for new movies and series is made directly after the cleanup, the files and series that were previously deleted will be found again.

Previously I had used an RPI4 with LibreELEC and this had the same problem after a while. I then completely reinstalled the system and it worked fine for a while.
Same now on the Ugoos, as well.

### To Reproduce

1. Cleanup DB
2. Scan for new Content

### Debuglog

The debuglog can be found here:
Will have to create one.

### Your Environment
Hardware:
Ugoos AM6+
CoreELEC 21.1.1 stable/nightly
Movies/Series are on a SMB share

**Comment by 78andyp** [comment]
Are these what is deleted??

```
2024-09-11 14:42:56.325 T:3954    error <general>: SMBDirectory->GetDirectory: Unable to open directory : 'smb://USERNAME:PASSWORD@192.168.yyy.xxx/freigabe/MedienDB/FILME/REMUX%20(HD)'
                                                   unix_err:'2' error : 'No such file or directory'
2024-09-11 14:42:56.325 T:3954    error <general>: GetDirectory - Error getting smb://smb01/freigabe/MedienDB/FILME/REMUX (HD)/
2024-09-11 14:42:57.367 T:3954    error <general>: SMBDirectory->GetDirectory: Unable to open directory : 'smb://USERNAME:PASSWORD@192.168.yyy.xxx/puffer/SERIEN/My%20Hero%20Academia%20(2016)%20(65930)'
                                                   unix_err:'2' error : 'No such file or directory'
2024-09-11 14:42:57.367 T:3954    error <general>: GetDirectory - Error getting smb://smb01/puffer/SERIEN/My Hero Academia (2016) (65930)/
2024-09-11 14:42:57.679 T:3954    debug <general>: CUtil::GetMatchingSource: no matching source found for [videodb://movies/sets/213/?setid=213]
```

**Comment by xadox-1st** [comment]
No. And the named folders are still there and accessable for Kodi, as well.

**Comment by DaVukovic** [comment]
>A cleanup of the library also removes movies and series from sources that are available.
If a scan for new movies and series is made directly after the cleanup, the files and series that were previously deleted will be found again.

Can these files, which have been added after the scan, be played? I just want to be sure that those "files" aren't leftovers from the library. 

Have you already asked the CoreELEC guys and are you able to reproduce with a current LibreELEC version?

And please stop modifying the logfile. There's no need to redact your IP as that's a private IP. Username and passwords are redacted already automatically. Nobody can do anything with that IP. The IP from my Kodi client is 192.168.1.162 and from my NAS is 192.168.1.175 and you can't do anything with that info ;)
